src/tigerbeetle-java/src/main/java/com/tigerbeetle/Transfers.java

```diff
@@ -0,0 +1,460 @@
+     * <p>
+     * Must be zero if this is not a {@link TransferFlags#POST_PENDING_TRANSFER post} or
+     * {@link TransferFlags#VOID_PENDING_TRANSFER post} transfer. Null values are converted to zero.
+     *
+     * @param pendingId an array of 16 bytes representing the 128-bit value.
+     * @throws IllegalArgumentException if {@code pendingId} is not 16 bytes long.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     */
+    public void setPendingId(final byte[] pendingId) {
+        putUInt128(at(Struct.PendingId), pendingId);
+    }
+
+    /**
+     * Gets the the interval (in nanoseconds) after a {@link TransferFlags#PENDING pending}
+     * transfer's creation that it may be posted or voided.
+     *
+     * @return a 64-bit integer.
+     */
+    public long getTimeout() {
+        return getUInt64(at(Struct.Timeout));
+    }
+
+    /**
+     * Sets the the interval (in nanoseconds) after a {@link TransferFlags#PENDING pending}
+     * transfer's creation that it may be posted or voided.
+     * <p>
+     * Must be zero if this is not a {@link TransferFlags#PENDING pending} transfer.
+     *
+     * @param timeout A 64-bit integer.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     */
+    public void setTimeout(final long timeout) {
+        putUInt64(at(Struct.Timeout), timeout);
+    }
+
+    /**
+     * Gets an identifier used to enforce that transfers must be between accounts of the same
+     * {@link Accounts#setLedger ledger}.
+     *
+     * @return a 32-bit integer.
+     */
+    public int getLedger() {
+        return getUInt32(at(Struct.Ledger));
+    }
+
+    /**
+     * Sets an identifier used to enforce that transfers must be between accounts of the same
+     * {@link Accounts#setLedger ledger}.
+     *
+     * @param ledger a 32-bit integer.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     */
+    public void setLedger(final int ledger) {
+        putUInt32(at(Struct.Ledger), ledger);
+    }
+
+    /**
+     * Gets a user-defined enum denoting the reason for (or category of) the transfer.
+     *
+     * @return a 16-bit unsigned integer.
+     */
+    public int getCode() {
+        return getUInt16(at(Struct.Code));
+    }
+
+    /**
+     * Sets a user-defined enum denoting the reason for (or category of) the transfer.
+     * <p>
+     * Must be non-zero.
+     *
+     * @param code a 16-bit unsigned integer defined by the user.
+     * @throws IllegalArgumentException if code is negative or greater than 65535.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     */
+    public void setCode(final int code) {
+        putUInt16(at(Struct.Code), code);
+    }
+
+    /**
+     * Gets a bitfield that specifies (optional) transfer behavior.
```

Comment:
These especially need links to the main documentation for the user to understand the flags.

Reply:
Constants for the user to build the flags field would be great too. So they can just `Flags.post_pending_transfer | Flags.whatever`